### PR TITLE
[Bug] fix raise error bug in registering multiple names

### PR DESCRIPTION
**Pull request by nbei**

### CODE REVIEW

**Comment by zhouzaida** [comment]
ping @xvjiarui @ZwwWayne

File: mmcv/utils/registry.py [hunk 1]
```diff
@@ -304,8 +300,10 @@ def register_module(self, name=None, force=False, module=None):
             return module
 
         # raise the error ahead of time
-        if not (name is None or isinstance(name, str)):
-            raise TypeError(f'name must be a str, but got {type(name)}')
+        if not (name is None or isinstance(name, str) or is_seq_of(name, str)):
+            raise TypeError(
+                'name must be either of None, an instance of str or list,'
```
Comment (ZwwWayne) [inline mmcv/utils/registry.py:305]: it could also be a tuple?

File: mmcv/utils/registry.py [hunk 2]
```diff
@@ -304,8 +300,10 @@ def register_module(self, name=None, force=False, module=None):
             return module
 
         # raise the error ahead of time
-        if not (name is None or isinstance(name, str)):
-            raise TypeError(f'name must be a str, but got {type(name)}')
+        if not (name is None or isinstance(name, str) or is_seq_of(name, str)):
```
Comment (hellock) [inline mmcv/utils/registry.py:303]: `_register_module()` is also called at Line298.

**Comment by hellock** [comment]
What is the intention of this PR?

**Comment by nbei** [comment]
> What is the intention of this PR?

Since we have supported a sequence of names, the logic for `raising error` should be modified.

**Comment by ZwwWayne** [comment]
Unit test failed.

**Comment by ZwwWayne** [comment]
ping @nbei

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**